omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BaseDelegatingResource.java

```diff
@@ -441,6 +444,46 @@ private void maybeDecorateWithType(SimpleObject simple, T delegate) {
 		if (hasTypesDefined())
 			simple.add(RestConstants.PROPERTY_FOR_TYPE, getTypeName(delegate));
 	}
+
+	/**
+	 * If this resource is an Order, add a STATUS field to the REST response which indicates whether the order is ACTIVE or INACTIVE.
+	 *
+	 * @param simple simplified representation which will be decorated with the status
+	 * @param delegate the object that simple represents
+	 */
+
+	private void maybeDecorateWithStatus(SimpleObject simple, T delegate) {
```

Comment:
Is there a reason to have this kind of class-specific logic here instead of in, say, an order-focused endpoint? This seems very hacky...